src/main/java/org/openbase/bco/bcozy/controller/powerterminal/chartcontroller/TreeChartController.java

```diff
@@ -35,14 +36,17 @@ public TreeChartController() {
     public void init(ChartStateModel chartStateModel, PowerChartVisualizationController powerChartVisualizationController) {
         view = new ImageView();
         view.setPreserveRatio(true);
-        view.setFitHeight(Screen.getPrimary().getVisualBounds().getHeight()/1.2);
+        view.setFitHeight(Screen.getPrimary().getVisualBounds().getHeight() / 1.2);
     }
 
     @Override
     public ScheduledFuture enableDataRefresh(long interval, ChartStateModel chartStateModel) {
         ScheduledFuture refreshSchedule = null;
         try {
-            refreshSchedule = GlobalScheduledExecutorService.scheduleAtFixedRate(() -> Platform.runLater(() -> updateChart(chartStateModel)),
+            refreshSchedule = GlobalScheduledExecutorService.scheduleAtFixedRate(() -> {
+                        List<ChartData> data = PowerTerminalDBService.getAverageConsumption(chartStateModel.getDateRange(), chartStateModel.getSelectedConsumer());
```

Comment:
To improve the performance, check if the tree pane is currently displayed and only then perform the update task. Otherwise, even when only the location map is displayed, the db is continuously queried.

Reply:
Fixed in 31576ec